src/directives/markers.js

```diff
@@ -114,10 +116,12 @@ angular.module("leaflet-directive").directive('markers',
             }
         }
     };
-    var _destroy = function(markerModels, oldMarkerModels, lMarkers, map, layers){
-        // Delete markers from the array
+    var _seeWhatWeAlreadyHave = function(markerModels, oldMarkerModels, lMarkers, isEqual, cb){
```

Comment:
This function is mostly what _destroy did and is the functionality that I said needed to be in _addMarkers (called before). The difference being that on adding it looks for matches and on _destroy it looks for differences.